image/image.go

```diff
@@ -113,6 +113,13 @@ func localSnaps(tsto *ToolingStore, opts *Options) (*localInfos, error) {
 				info.Revision = si.Revision
 				info.Channel = opts.Channel
 			}
+			// ensure the kernel-track is honored
+			if model.Kernel() != "" && model.KernelTrack() != "" {
+				info.Channel, err = makeKernelChannel(model.KernelTrack(), opts.Channel)
```

Comment:
I'm missing something,  are setting to this channel for all snaps? that seems wrong